src/views/blackout/Blackouts.vue

```diff
@@ -69,47 +130,81 @@
 import Context from '@/mixins/Context'
 import PageTitle from '@/components/util/PageTitle'
 import Utils from '@/mixins/Utils'
-import {deleteBlackout, getAllBlackouts} from '@/api/blackout'
+import {createBlackout, deleteBlackout, getAllBlackouts} from '@/api/blackout'
 
 export default {
   name: 'Blackouts',
   mixins: [Context, Utils],
   components: {PageTitle},
   data: () => ({
+    blackouts: undefined,
+    dialog: undefined,
     headers: [
       {text: 'Name', value: 'name'},
       {text: 'Start Date', value: 'startDate'},
       {text: 'End Date', value: 'endDate'},
       {text: 'Created', value: 'createdAt'},
       {text: 'Delete', class: 'pl-5 pr-0 mr-0'}
     ],
-    blackouts: undefined,
+    isSaving: false,
+    name: undefined,
+    range: undefined,
     refreshing: false
   }),
-  mounted() {
+  computed: {
+    disableSave() {
+      return !this.$_.trim(this.name) || !this.range.start || !this.range.end
+    }
+  },
+  created() {
     this.$loading()
-    this.loadAllBlackouts().then(() => {
+    this.reset()
+    getAllBlackouts().then(data => {
+      this.blackouts = data
       this.$ready('Blackouts')
     })
   },
   methods: {
+    cancel() {
+      this.reset()
+      this.alertScreenReader('Cancelled.')
+    },
     createBlackout() {
-      this.goToPath('/blackout')
+      this.isSaving = true
+      const format = date => {
+        return this.$moment(date).format('YYYY-MM-DD')
+      }
+      createBlackout(this.name, format(this.range.start), format(this.range.end)).then(() => {
+        getAllBlackouts().then(data => {
+          this.blackouts = data
+          this.alertScreenReader('Blackout created')
+          this.reset()
+          this.isSaving = false
+        })
+      })
     },
     deleteBlackout(blackoutId) {
       this.refreshing = true
       deleteBlackout(blackoutId).then(() => {
         this.alertScreenReader('Blackout deleted.')
-        this.loadAllBlackouts().then(() => {
-          this.refreshing = false
-        })
       })
     },
-    loadAllBlackouts() {
-      return getAllBlackouts().then(data => {
-        this.blackouts = data
-      })
+    reset() {
+      this.dialog = false
+      this.range = {
+        start: undefined,
+        end: undefined,
+      }
     }
   }
 }
 </script>
+
+<style scoped>
+.add-date-btn {
+}
+.modify-date-btn {
+}
+.remove-date-btn {
+}
+</style>
```

Comment:
@johncrossman Are these empty styles meant to be here?

Reply:
No they're not.  :)  I'll remove.